src/AutoRest.CSharp/MgmtTest/Models/JsonRawValue.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Input;
+
+namespace AutoRest.CSharp.MgmtTest.Models
+{
+    internal readonly struct JsonRawValue
+    {
+        public readonly object? RawValue;
+
+        public JsonRawValue(object? rawValue)
+        {
+            RawValue = rawValue;
+        }
+
+        public bool IsNull()
+        {
+            return RawValue is null;
+        }
+
+        public bool IsEnumerable()
+        {
+            if (RawValue == null)
+                return false;
+            return typeof(IEnumerable<object>).IsAssignableFrom(RawValue.GetType());
+        }
+        public IEnumerable<object> AsEnumerable()
+        {
+            return RawValue is null ? Enumerable.Empty<object>() : (IEnumerable<object>)RawValue;
+        }
+
+        public bool IsString()
+        {
+            if (RawValue == null)
+                return false;
+            return RawValue is string;
+        }
+        public string? AsString()
+        {
+            return RawValue?.ToString();
+        }
```

Comment:
If we always expect to use this method like this:
```
if (v.IsString())
{
var s = v.AsString();
}
```
we could just return a `string` instead of nullable `string?` here